app/dockerdwrapper.c

```diff
@@ -533,10 +774,59 @@ stop_dockerd(void)
     syslog(
         LOG_ERR, "Failed to send SIGKILL to child. Error: %s", strerror(errno));
   }
+
 end:
   return killed;
 }
 
+/**
+ * @brief Start fcgi and dockerd.
+ *
+ * @return exit_code. 0 if successful, -1 otherwise
+ */
+static int
+start(void) {
+  bool use_sdcard = false;
+  bool use_tls = false;
+  bool use_ipc_socket = false;
+  bool use_verbose = false;
+  syslog(LOG_INFO,"start called");
+
+  if (!get_verbose_selection(&use_verbose)) {
+    syslog(LOG_INFO, "Failed to get verbose selection. Uninstall and reinstall the acap?");
+    exit_code = -1;
+    goto end;
+  }
+  if (!get_and_verify_tls_selection(&use_tls)) {
+    syslog(LOG_INFO, "Failed to verify tls selection");
+    goto fcgi; /* do not start dockerd */
+  }
+  if (!get_ipc_socket_selection(&use_ipc_socket)) {
+    syslog(LOG_INFO, "Failed to get ipc socket selection");
+    goto fcgi; /* do not start dockerd */
+  }
+  if (!get_and_verify_sd_card_selection(&use_sdcard)) {
+    syslog(LOG_INFO, "Failed to setup sd_card");
+    goto fcgi; /* do not start dockerd */
+  }
```

Comment:
should we move these to before the get_and_verify_tls_selection check and send them to end if they fail

Reply:
Think it makes sense to treat IPC soceket seletction the same way.. but better to delay verifying the SD card selection for as long as possible?